### PR TITLE
Fix multiple files spilled for the distinct hash table

### PR DESCRIPTION
**Pull request by xiaoxmeng**
The existing distinct aggregation implementation assumes that there is one file generated
for each spill run. And use stream id 0 to detect if a row read from spilled file is distinct one
or not. This is no longer true after we add support to configure the max number of rows to
spill in each sorted spill file for aggregation which means stream id > 0 could also contains
the distinct values. This will cause incorrect data result and reported by [issue](https://github.com/facebookincubator/velox/issues/9219).

This PR fixes this issue by recording the number of spilled files on the first spill in grouping
set to detect the spilled files that contain the seen distinct values. Unit test is added to reproduce
and verify the fix. Also removed the unused spill config

### CODE REVIEW

**Review by duanmeng — APPROVED** [review]
Looks great % some nits

File: velox/exec/GroupingSet.h [hunk 1]
```diff
@@ -326,6 +326,10 @@ class GroupingSet {
   bool remainingMayPushdown_;
 
   std::unique_ptr<Spiller> spiller_;
+  // Sets to the number of files stores the spilled distinct hash table which
+  // are the files generated by the first spill call. This only applies for
+  // distinct hash aggregation.
+  size_t numDistinctTableSpilledFiles_{0};
```
Comment (duanmeng) [inline velox/exec/GroupingSet.h:332]: s/numDistinctTableSpilledFiles_/numDistinctSpilledFiles_/g ?

File: velox/exec/Spill.cpp [hunk 2]
```diff
@@ -155,6 +155,17 @@ void SpillState::finishFile(uint32_t partition) {
   writer->finishFile();
 }
 
+size_t SpillState::numFinishedFiles(uint32_t partition) const {
+  if (!isPartitionSpilled(partition)) {
+    return 0;
+  }
+  auto* writer = partitionWriter(partition);
```
Comment (duanmeng) [inline velox/exec/Spill.cpp:162]: const auto?

**Review by zhztheplayer — APPROVED** [review]
Thanks for the fix!

File: velox/core/QueryConfig.h [hunk 3]
```diff
@@ -241,15 +241,6 @@ class QueryConfig {
...
   static constexpr const char* kMaxSpillFileSize = "max_spill_file_size";
 
-  /// The min spill run size limit used to select partitions for spilling. The
-  /// spiller tries to spill a previously spilled partitions if its data size
-  /// exceeds this limit, otherwise it spills the partition with most data.
-  /// If the limit is zero, then the spiller always spill a previously spilled
-  /// partition if it has any data. This is to avoid spill from a partition with
-  /// a small amount of data which might result in generating too many small
-  /// spilled files.
-  static constexpr const char* kMinSpillRunSize = "min_spill_run_size";
```
[the remark this answers is not in the document]
Comment (xiaoxmeng) [reply velox/core/QueryConfig.h:251]: I will remove this in followup after Prestissimo has removed the config reference

File: velox/exec/fuzzer/AggregationFuzzerBase.cpp [hunk 4]
```diff
@@ -403,8 +407,12 @@ velox::test::ResultOrError AggregationFuzzerBase::execute(
       spillDirectory = exec::test::TempDirectoryPath::create();
       builder.spillDirectory(spillDirectory->path)
           .config(core::QueryConfig::kSpillEnabled, "true")
-          .config(core::QueryConfig::kAggregationSpillEnabled, "true");
-      spillPct = 100;
+          .config(core::QueryConfig::kAggregationSpillEnabled, "true")
+          .config(
+              core::QueryConfig::kMaxSpillRunRows,
+              std::to_string(randInt(32, 1L << 30)));
```
[the remark this answers is not in the document]
Comment (xiaoxmeng) [reply velox/exec/fuzzer/AggregationFuzzerBase.cpp:413]: Thanks!

**Pull request closed without merging**